ckan/logic/action/create.py

```diff
@@ -295,6 +296,13 @@ def resource_create(context, data_dict):
         pkg_dict['resources'] = []
 
     upload = uploader.get_resource_uploader(data_dict)
+    ## Get out resource_id resource from model as it will not appear in
+    ## package_show until after commit
+    upload.upload(context['package'].resources[-1].id,
+                  uploader.get_max_resource_size())
+
+    data_dict['mimetype'] = mimetypes.guess_type(data_dict['url'])[0] or upload.mimetype
```

Comment:
We need to check if the user has provided a `mimetype` or `size` in the data_dict before overriding them